libminifi/include/core/ParameterTokenParser.h

```diff
@@ -131,5 +138,27 @@ class ParameterTokenParser {
   std::vector<std::unique_ptr<ParameterToken>> tokens_;
 };
 
+class NonSensitiveParameterTokenParser : public ParameterTokenParser {
+ public:
+  using ParameterTokenParser::ParameterTokenParser;
+
+ protected:
+  std::string getRawParameterValue(const Parameter& parameter) const override;
+};
+
+class SensitiveParameterTokenParser : public ParameterTokenParser {
+ public:
+  SensitiveParameterTokenParser(std::string input, utils::crypto::EncryptionProvider& sensitive_values_encryptor)
+      : ParameterTokenParser(std::move(input)),
+        sensitive_values_encryptor_(sensitive_values_encryptor) {
+  }
+
+ protected:
+  std::string getRawParameterValue(const Parameter& parameter) const override;
+
+ private:
+  utils::crypto::EncryptionProvider& sensitive_values_encryptor_;
+};
+
```

Comment:
I think inheritance is unnecessary abstraction here. Wherever we're parsing the property could just call the token parser with or without decryption as needed. It's just an `if` statement. If encrypted, then decrypt. Then parse and substitute parameters.

Reply:
I think this is a cleaner solution for the following reasons:
- We need information about the sensitivity of the handled property using parameters to see if its sensitivity and the sensitivity of the parameters match up which also needs to be passed down. That's multiple conditions to check which is abstracted and handled by the polymorphism so it's easier to read.
- The EncryptionProvider is only needed for sensitive properties, so in case we are handling non-sensitive properties it's unnecessary to create an EncryptionProvider to pass to the token parser if we already know it will be unused.

Reply:
I still think that it could probably be done in a simpler way, and the abstraction adds complexity, but this is not a hill I want to die on, it's fine.